### ISSUE TITLE
dbg_macro ignored in macro call sites

### ISSUE THREAD

**Issue by Wilfred** [rust-lang/rust-clippy#12131]
### Summary

When `dbg!` is used inside another macro call, the dbg_macro lint doesn't fire.

### Lint Name

dbg_macro

### Reproducer

I tried this code:

```rust
#![warn(clippy::dbg_macro)]

fn foo(s: &str) {
    dbg!(s); // lint
}

fn foo2(s: &str) {
    format!("{}.rs", dbg!(s)); // no lint
}
```

I expected to see this happen: Both occurrences of `dbg!` produce a lint.

Instead, this happened: Only `foo` gets the lint.
